README.md

```diff
@@ -57,3 +57,92 @@ Releases are automatically pushed on commits merged to master that fulfill the f
 The release process is building all runtimes and then puts them into a release in this github repository.
 
 The format of [`CHANGELOG.md`](CHANGELOG.md) is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+
+# Weight Generation
+
+To generate weights for a runtime
+
+1. Build `chain-spec-generator` with `--profile production --features runtime-benchmarks`
+2. Use it to build a chain spec for your runtime, e.g. `./target/production/chain-spec-generator --raw polkadot-local > polkadot-chain-spec.json`
+3. Create `file_header.txt`
+
+```text
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+```
+
+4. `rsync` chain spec/s and the file header to a benchmark machine
+
+5. Build `polkadot-sdk` with `--profile production --features runtime-benchmarks` on the benchmark machine
+
+6. Create output directories for the weights on the benchmark machine
+
+7. Run on the benchmark machine:
+
+```bash
+#!/bin/bash
+
+# Default value is 'polkadot', but you can override it by passing a different value as an argument
+CHAIN=${1:-polkadot}
+
+pallets=($(
+  ./target/production/polkadot benchmark pallet --list \
+    --chain=./$CHAIN-chain-spec.json |
+    tail -n+2 |
+    cut -d',' -f1 |
+    sort |
+    uniq
+);
+
+for pallet in "${pallets[@]}"; do
+  echo "Running benchmarks for $pallet"
+  ./target/production/polkadot benchmark pallet \
+    --chain=./$CHAIN-chain-spec.json \
+    --steps=50 \
+    --repeat=20 \
+    --pallet=$pallet \
+    --extrinsic=* \
+    --wasm-execution=compiled \
+    --heap-pages=4096 \
+    --output=./$CHAIN-weights \
+    --header=./file_header.txt
+done
+```
+
+You probably want to do this inside a `tmux` session or similar, as it will take a while.
+
+8. `rsync` the weights back to your local machine, replacing the existing weights.
+
+9. Manually fix XCM weights by
+- Replacing `impl<T: frame_system::Config> xxx::yyy::WeightInfo<T> for WeightInfo<T> {` with `impl<T: frame_system::Config> WeightInfo<T> {`
+- Marking all functions `pub(crate)`
+- Removing any unused functions
+
+10. Commit the weight changes.
+
+11. If not installed, `cargo install subweight`, and check the weight changes with `subweight compare commits --path-pattern "./relay/polkadot/src/weights/*.rs" --method asymptotic --ignore-errors HEAD HEAD^1`. Ensure the changes are reasonable.
```

Comment:
```suggestion
11. If not installed, `cargo install subweight`, and check the weight changes with `subweight compare commits --path-pattern "./**/weights/*.rs" --method asymptotic --ignore-errors HEAD origin/main`. Ensure the changes are reasonable.
```
So that it compares all chains, in case more than just Polkadot was updated.